src/omnibar.js

```diff
@@ -67,10 +68,13 @@ async function makeSuggestion(query, suggest) {
 }
 
 const acceptInput = (text, disposition) => {
-    // TODO if text is not a suggested URL, open the overview with this query.
+    // Checks whether the text is a suggested url
+    const validUrl = tldjs.isValid(text)
+    const overviewPageWithQuery = '/overview/overview.html?searchQuery=' + text
+
     switch (disposition) {
         case 'currentTab':
-            browser.tabs.update({url: text})
+            browser.tabs.update({url: validUrl ? text : overviewPageWithQuery})
             break
         case 'newForegroundTab':
```

Comment:
Should this also do the same as current tab. When is the disposition set to "newForegroundTab" or "newBackgroundTab"? Anything I tried it was always set to "currentTab".